etna/collections/models.py

```diff
@@ -145,12 +150,38 @@ def related_articles(self):
         from etna.articles.models import ArticlePage
 
         return (
-            ArticlePage.objects.filter(topic=self)
-            .live()
-            .select_related("teaser_image")
-            .order_by("title")[:3]
+            ArticlePage.objects.live()
+            .public()
+            .filter(id__in=self.related_page_ids)
+            .order_by("-first_published_at")
+            .exclude(title=self.featured_article)
+        )
+
+    @cached_property
+    def related_record_articles(self):
+        from etna.articles.models import RecordArticlePage
+
+        return (
+            RecordArticlePage.objects.live()
+            .public()
+            .filter(id__in=self.related_page_ids)
+            .order_by("-first_published_at")
+            .exclude(title=self.featured_record_article)
+        )
+
+    @cached_property
+    def related_highlight_galleries(self):
```

Comment:
Note: let's use the naming convention of `related_highlight_gallery_pages`, we're getting the highlight gallery pages, not specifically the gallery, it's semantics but I think it's clearer, what do you think?